### ISSUE TITLE
pretraing path issues

### ISSUE THREAD

**Issue by chrisgwe** [MaitySubhajit/SelfDocSeg#2]
python SelfDocSeg-main/pretraining/train_ssl.py --num_eval_classes 16 --dataset_name DocLayNet --knn_train_root F:\ComputerVision\Academicresearch\Examples\SelfDocSeg-main\SelfDocSeg-main\dataset\DocLayNet\PNG\train.json --knn_eval_root SelfDocSeg-main\dataset\DocLayNet\PNG\COCO\test.json --dataset_root SelfDocSeg-main\dataset\DocLayNet\PNG --logs_root ./benchmark_logs --num_workers 0 --max_epochs 800 --batchsize 8 --n_runs 1 --learning_rate 0.2 --lr_decay 5e-4 --wt_momentum 0.99 --bin_threshold 239 --kernel_shape rect --kernel_size 3 --kernel_iter 2 --eeta 0.001 --alpha 1.0 --beta 1.0
Total Images :  81475

could you tell me why this issue The directory name is invalid: 'F:\\ComputerVision\\Academicresearch\\Examples\\SelfDocSeg-main\\SelfDocSeg-main\\dataset\\DocLayNet\\PNG\\train.json'
<img width="1667" alt="Screenshot 2023-10-09 183359" src="https://github.com/MaitySubhajit/SelfDocSeg/assets/123208317/6c8564bc-bc0c-4c03-9f8c-fbbfb475cd14">

**Comment by chrisgwe** [comment]
please help!

**Comment by MaitySubhajit** [comment]
Hello @chrisgwe !

**This is the expected behaviour of the code.**

For kNN linear evaluation, you'll need any document image classification dataset, that too in a particular format. As _DocLayNet_ is a document layout analysis dataset, and not a classification one, it does not have any single class label associated with a single image. The _.json_ files contain bounding box-level annotation, while the code expects only directories with class names, each containing all the images for that particular class.

For the document classification dataset, the current code in this repository requires it in the PyTorch ImageFolder format (Refer to [Dataset section](https://github.com/MaitySubhajit/SelfDocSeg#Dataset) for detailed information). We have used the _RVL-CDIP_ dataset for this purpose. _Tobacco_ dataset can also be used for the same.

**Further Notes:**

1. You can override the ImageFolder format by using a standard PyTorch dataset with the same transformation. The Lightly Trainer expects the dataset to pass _tuple(image, label, image file name)_ during training.
2. The best course of action is to prepare a dataset with bounding box-level crops of all the objects in the _DocLayNet_ dataset, meaning a classification dataset containing all the 11 classes in DocLayNet with all possible layout object instances per class.

Let me know if this helps!
Thanks!

**Comment by chrisgwe** [comment]
Dear Maity 
I hope this find you well!

I try to pretraining the train_ssl.py , but the issue I faceing right now is the train_root alwasy shows invalid. I try to follow the Dataset section the root directory to put rvl-cdip dataset. Always shows same error. could you tell me where is the problem with my directory, thank you so much!! 

python SelfDocSeg-main/pretraining/train_ssl.py --num_eval_classes 16 --dataset_name rvl-cdip --knn_train_root SelfDocSeg-main\dataset\train.txt --knn_eval_root SelfDocSeg-main\dataset\test.txt --dataset_root SelfDocSeg-main\dataset\rvl-cdip --logs_root ./benchmark_logs --num_workers 0 --max_epochs 800 --batchsize 8 --n_runs 1 --learning_rate 0.2 --lr_decay 5e-4 --wt_momentum 0.99 --bin_threshold 239 --kernel_shape rect --kernel_size 3 --kernel_iter 2 --eeta 0.001 --alpha 1.0 --beta 1.0

<img width="194" alt="image" src="https://github.com/MaitySubhajit/SelfDocSeg/assets/123208317/364a351c-c126-453a-b5d0-4eaa736fadaf">

<img width="188" alt="image" src="https://github.com/MaitySubhajit/SelfDocSeg/assets/123208317/40eeac2e-b6e7-462f-adb2-36af9577d076">

**Comment by chrisgwe** [comment]
<img width="1688" alt="image" src="https://github.com/MaitySubhajit/SelfDocSeg/assets/123208317/bc66249d-50b8-4f38-8e2b-089001972895">

**Comment by chrisgwe** [comment]
I think this error is related to the path with my train.txt

**Comment by MaitySubhajit** [comment]
Yes, the error is related to train.txt and test.txt and as I have mentioned earlier, the arguments --knn_train_root and --knn_eval_root expect two paths where it will have directories with class names, each containing all the instances of that class. Instead, it is pointing to .txt files in your case.

Also, the directory structure you have shown is incorrect from the code perspective. The --knn_train_root or --knn_eval_root should point to the paths each, where 16 directories are present representing 16 classes of RVL-CDIP and each of the 16 directories contain corresponding image instances.

**Comment by MaitySubhajit** [comment]
Note: RVL-CDIP is not directly available in the required ImageFolder structure as required by the code from the Huggingface link. Either you have to modify it to ImageFolder structure, or you can override the same with a custom PyTorch Dataset class to pass the image, class label, and image file name when the data loader is called.

We may release a script to convert RVL-CDIP to ImageFolder format later on.

**Comment by chrisgwe** [comment]
quote"We may release a script to convert RVL-CDIP to ImageFolder format later on." this answers my concern, appreacite mate!

**Comment by chrisgwe** [comment]
looking forward to that release

**Comment by chrisgwe** [comment]
Just want to ask when will you put the script for convert dataset

**Comment by MaitySubhajit** [comment]
Hi!

Unfortunately, we are busy with several other projects and thus there will be a delay in preparing this script. Please do expect it by the end of the next month.

In the meantime, you can overwrite the dataset classes with custom Torchvision datasets for the RVL-CDIP format available to you.

Thanks!

**Comment by Cyc-striving** [comment]
> Note: RVL-CDIP is not directly available in the required ImageFolder structure as required by the code from the Huggingface link. Either you have to modify it to ImageFolder structure, or you can override the same with a custom PyTorch Dataset class to pass the image, class label, and image file name when the data loader is called.
> 
> We may release a script to convert RVL-CDIP to ImageFolder format later on.

Dear Maity,I want to ask when will you put the script for convert dataset
